samples/TagHelperPack.Sample/Pages/Index.cshtml

```diff
@@ -61,7 +63,7 @@
             </div>
             <div class="form-group">
                 <label asp-for="Customer.FirstName"></label>
-                <editor for="Customer.FirstName" />
+                <editor for="Customer.FirstName" view-data="@(new Dictionary<string, object>{ ["color"] = "red" })" view-data-font-style="@("italic")" />
```

Comment:
So I found this is confusing: it doesn't like just `view-data-font-style="italic"` here.

> The name 'italic' does not exist in the current context

Is that because I used `IDictionary<string, object>` ([like `ComponentTagHelper`](https://github.com/dotnet/aspnetcore/blob/b39a258cbce1b16ee98679ef7d2ddc2e09040a6b/src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs#L35-L36)) instead of `IDictionary<string, string>` ([like `AnchorTagHelper`](https://github.com/dotnet/aspnetcore/blob/b39a258cbce1b16ee98679ef7d2ddc2e09040a6b/src/Mvc/Mvc.TagHelpers/src/AnchorTagHelper.cs#L136-L137))?

Reply:
I'd have to refresh my memory on the rules when using dictionary suffix properties. It's unusual that you'd need that much escaping though, I'd expect `view-data-font-style="italic"` to work.

Reply:
OK I tried this and I think the cause is exactly what you said. I'd change this `IDictionary<string, string>` if the intent is to allow string values to be passed through but if you want `object` supported as a value to pass through then it will need to stay as `IDictionary<string, string>`. Ultimately the underlying API just accepts `object` so there isn't much that can be done.

Reply:
I think supporting `object` makes more sense, since that aligns with the underlying API.

Is this worth filing a Razor bug?

Reply:
The only way I could see this working via a change in Razor would be to change the default context chosen by the compiler once inside the attribute quotes for attributes mapped to object values, i.e. it would need to change to HTML instead of code (C#) which it is today. Any literal value entered would be interpreted as a string, and to pass anything from code would require an '@' prefixed Razor expression. That would of course be a breaking change (as it's the opposite of today) though so isn't something that can be done simply (if at all).